analytical_engine/core/app/pregel/pregel_vertex.h

```diff
@@ -46,41 +46,49 @@ class PregelVertex {
 
   PregelVertex() = default;
 
-  std::string id() { return std::to_string(fragment_->GetId(vertex_)); }
+  virtual std::string id() { return std::to_string(fragment_->GetId(vertex_)); }
```

Comment:
Why set all those methods to virtual? Are they all necessary?

Reply:
remove unnecessary virtual.